src/arithmetic/arithmetic_expression.c

```diff
@@ -267,8 +274,11 @@ SIValue AR_EXP_Evaluate(AR_ExpNode *root, const Record r) {
 					char *error;
 					SIValue v = Record_GetScalar(r, root->operand.variadic.entity_alias_idx);
 					asprintf(&error, "Type mismatch: expected a map but was %s", SIType_ToString(SI_TYPE(v)));
```

Comment:
Consider condensing this entire logic into a single function which accepts and error message sets the error and jumps.

Reply:
I had some vague concerns about stack unwinding when adding another function call, but I guess that doesn't really make sense! Will do.

Reply:
In such cases you might want to consider `inline`

Reply:
`inline` is a compiler hint, not a mandate - to my knowledge, the only way to guarantee inlining is to `static inline` in header files (like `rmalloc.h`).